app/presenters/research_output_presenter.rb

```diff
@@ -54,22 +54,23 @@ def byte_sizable?
   end
 
   # Returns the options for subjects for the repository filter
+  # rubocop:disable Metrics/AbcSize
   def self.selectable_subjects
     [
-      '23-Agriculture, Forestry, Horticulture and Veterinary Medicine',
-      '21-Biology',
-      '31-Chemistry',
-      '44-Computer Science, Electrical and System Engineering',
-      '45-Construction Engineering and Architecture',
-      '34-Geosciences (including Geography)',
-      '11-Humanities',
-      '43-Materials Science and Engineering',
-      '33-Mathematics',
-      '41-Mechanical and industrial Engineering',
-      '22-Medicine',
-      '32-Physics',
-      '12-Social and Behavioural Sciences',
-      '42-Thermal Engineering/Process Engineering'
+      "23-#{_('Agriculture, Forestry, Horticulture and Veterinary Medicine')}",
+      "21-#{_('Biology')}",
+      "31-#{_('Chemistry')}",
+      "44-#{_('Computer Science, Electrical and System Engineering')}",
+      "45-#{_('Construction Engineering and Architecture')}",
+      "34-#{_('Geosciences (including Geography)')}",
+      "11-#{_('Humanities')}",
+      "43-#{_('Materials Science and Engineering')}",
+      "33-#{_('Mathematics')}",
+      "41-#{_('Mechanical and industrial Engineering')}",
+      "22-#{_('Medicine')}",
+      "32-#{_('Physics')}",
+      "12-#{_('Social and Behavioural Sciences')}",
+      "42-#{_('Thermal Engineering/Process Engineering')}"
```

Comment:
I dont think the hashtag notation will work for this. In previous times we have used the format method so that gettext can identify the keys.  Translation gem is using gettext which uses the full string as a key in a hash object to look up the translations. so if a key is missing it wont translate.

So it should be something like ``_('23-Agriculture, Forestry, Horticulture and Veterinary Medicine')``.

Or you can try something like ```format("23-%{subject}", subject : _(Agriculture, Forestry, Horticulture and Veterinary Medicine'))``` which could make it more general. I am not 100% sure it works though.

Give it a try on the test project and talk with @200455939-yashu .

Reply:
<img width="885" alt="Screenshot 2023-08-01 at 3 00 58 PM" src="https://github.com/portagenetwork/roadmap/assets/71047780/547a0d49-2159-4089-82e1-75ea5a38a3fe">
<img width="426" alt="Screenshot 2023-08-01 at 3 00 17 PM" src="https://github.com/portagenetwork/roadmap/assets/71047780/160607c9-1c20-4516-926e-2b21b1f65c06">

Cool, good to double check. I just tried creating a couple translations in dmp-assistant-test. I guess the current code format is ok for translation.io to pick up.

Reply:
Actually, although it is getting picked up, this will still probably need to be fixed. When `.map do |subject|
      [subject.split('-').last, subject.gsub('-', ' ')]` is applied to the array, both elements in the array receive the translation. However, `subject.gsub('-', ' ')` should not be receiving the translation.